src/Storage/Controllers/DataController.cs

```diff
@@ -593,11 +595,11 @@ private async Task<ActionResult<DataElement>> InitiateDelayedDelete(Instance ins
             return Ok(updatedDateElement);
         }
 
-        private async Task<ActionResult<DataElement>> DeleteImmediately(Instance instance, DataElement dataElement)
+        private async Task<ActionResult<DataElement>> DeleteImmediately(Instance instance, DataElement dataElement, Application application)
```

Comment:
Reduce parameter to be only the storage container number value. No need for the full application object.

Reply:
Done